models/news-models.js

```diff
@@ -8,64 +12,119 @@ exports.selectTopics = () => {
-      return Promise.reject({status: 404, msg: "404 - No articles"}); 
-    }
-    articles.map(article => {
-      mappedArticles.push({author: article.author, title: article.title, article_id: article.article_id, topic: article.topic, created_at: article.created_at, votes: article.votes})
-    })
-    return mappedArticles;
-  });
+  return db
+    .query(`SELECT * FROM articles ORDER BY created_at DESC;`)
+    .then((result) => {
+      const articles = result.rows;
+      const mappedArticles = [];
+      if (articles.length === 0) {
+        return Promise.reject({ status: 404, msg: "404 - No articles" });
+      }
+      articles.map((article) => {
+        mappedArticles.push({
+          author: article.author,
+          title: article.title,
+          article_id: article.article_id,
+          topic: article.topic,
+          created_at: article.created_at,
+          votes: article.votes,
+        });
+      });
+      return mappedArticles;
+    });
 };
 
 exports.selectArticleById = (id) => {
   if (isNaN(id))
-    return Promise.reject({status: 400, msg: "400 - Does not match article ID type!"});
-  
-  return db.query(`SELECT * FROM articles WHERE article_id = ${id};`).then((result) => {
-    const article = result.rows;
-    if(article.length === 0) {
-      return Promise.reject({status: 404, msg: "404 - article not found"}); 
-    }
-    return result.rows;
-  })
+    return Promise.reject({
+      status: 400,
+      msg: "400 - Does not match article ID type!",
+    });
+
+  return db
+    .query(`SELECT * FROM articles WHERE article_id = ${id};`)
+    .then((result) => {
+      const article = result.rows;
+      if (article.length === 0) {
+        return Promise.reject({ status: 404, msg: "404 - article not found" });
+      }
+      return result.rows;
+    });
 };
 
-exports.selectComments = (id => {
+exports.selectComments = (id) => {
   if (isNaN(id))
-    return Promise.reject({status: 400, msg: "400 - article ID must be a number!"});
+    return Promise.reject({
+      status: 400,
+      msg: "400 - article ID must be a number!",
+    });
 
-    return db.query(`SELECT * FROM comments WHERE article_id = ${id};`).then((result) => {
+  return db
+    .query(`SELECT * FROM comments WHERE article_id = ${id};`)
+    .then((result) => {
       const comment = result.rows;
-      if(comment.length === 0) {
-        return Promise.reject({status: 404, msg: "404 - article not found"}); 
+      if (comment.length === 0) {
+        return Promise.reject({ status: 404, msg: "404 - article not found" });
       }
       return result.rows;
+    });
+};
+
+exports.postNewComment = (author, body, id) => {
+  if (isNaN(id)) {
+    return Promise.reject({ status: 400, msg: "400 - article unavailable!" });
+  }
+  const checkUsers = db.query(`SELECT * FROM users`);
+
+  return checkUsers
+    .then((checkUsers) => {
+      const users = checkUsers.rows;
+      let count = 0;
+
+      users.map((user) => {
```

Comment:
another misuse of `.map` (sorry)!

There are more appropriate methods to find out if something is in an array

look up `.some` 🔎